src/api/app/controllers/webui/package_controller.rb

```diff
@@ -92,22 +92,25 @@ def linking_packages
   end
 
   def dependency
-    unless Project.find_by_name(params[:dproject].to_s)
+    dep_project = Project.find_by_name(params[:dproject]) || Project.find_remote_project(params[:dproject]).try(:first)
+    unless dep_project
       flash[:error] = "Project '#{params[:dproject]}' is invalid."
       redirect_back(fallback_location: root_path)
       return
     end
 
     unless Architecture.archcache.include?(params[:arch])
       flash[:error] = "Architecture '#{params[:arch]}' is invalid."
-      redirect_back(fallback_location: project_show_path(project: params[:dproject]))
+      redirect_back(fallback_location: project_show_path(project: @project.name))
       return
     end
-    project_repositories = Project.find_by_name(params[:dproject]).repositories.pluck(:name)
+
+    # FIXME: It can't check repositories of remote projects
+    project_repositories = dep_project.repositories.pluck(:name)
```

Comment:
Well, this will crash then because `dep_project` is a string and `String` has no `.repositories`...

Reply:
`dep_project` is not a String. The `repositories` method doesn't deal with remote repositories.